Update players/spectators list in Initialize phase

danwang opened the pull request:
r? @rbtying 

Added a new display for users in the Initialize phase. Shows a tabular view rather than a side-by-side view. Buttons for moving/kicking/toggling observer mode are in each row

<img width="720" alt="Screen Shot 2020-04-19 at 1 13 09 AM" src="https://user-images.githubusercontent.com/935063/79680202-09c5cb80-81db-11ea-8398-5c29d4734144.png">


<img width="615" alt="Screen Shot 2020-04-19 at 1 15 00 AM" src="https://user-images.githubusercontent.com/935063/79680220-35e14c80-81db-11ea-8838-11792c94b33f.png">

rbtying commented on frontend/src/UserList.tsx:
```diff
@@ -0,0 +1,133 @@
+const UserRow = (props: RowProps) => (
+  <Row key={props.player.name}>
+    <Cell style={{width: '50%'}}>{props.player.name}</Cell>
+    <Cell style={{width: '50%'}}>Rank {props.player.level}</Cell>
```
What do you think about minimizing the space dedicated to the level (and skewing it more towards the player name).

Similarly, we can reduce the size of the overall table and increase readability that way -- the maximum player name length is 32 chars

rbtying replied:
This is less of an issue if the overall box is narrower as suggested below

danwang replied:
Changed "Rank" to be minimum width, so players column will eat up all available space

rbtying commented on frontend/src/UserList.tsx:
```diff
@@ -0,0 +1,133 @@
+);
+
+type Props = {
+  currentPlayer: IPlayer;
```
Can we still include the `landlord` information in this component? I don't think players will know to look in the dropdown in continuation settings

danwang replied:
Updated in fdcce4883c2a56bff0a886edc7aa3ccbbcf251fe

<img width="599" alt="Screen Shot 2020-04-19 at 3 00 22 AM" src="https://user-images.githubusercontent.com/935063/79681695-1140a100-81ea-11ea-820a-e33401451af7.png">

rbtying commented on frontend/src/UserList.tsx:
```diff
@@ -0,0 +1,133 @@
+import {WebsocketContext} from './WebsocketProvider';
+
+const Container = styled.div`
+  width: 80%;
```
perhaps, 40% with min-width at 500px or so? On wide screens the table is really wide right now

danwang replied:
Updated in 554cb908cd44258732970504a2f6c557f3306593

On wide screen, looks like this:

<img width="1439" alt="Screen Shot 2020-04-19 at 3 01 04 AM" src="https://user-images.githubusercontent.com/935063/79681684-084fcf80-81ea-11ea-973a-4cb436048bb1.png">

danwang commented:
ptal @rbtying 

Updated in554cb908cd44258732970504a2f6c557f3306593 and fdcce4883c2a56bff0a886edc7aa3ccbbcf251fe

what if we changed the Play phase player display from left-to-right to something like a circle? A ring could be helpful to see ordering if the player positions don't move from turn to turn. If so, there wouldn't need to be a change from vertical to horizontal

rbtying commented:
> what if we changed the Play phase player display from left-to-right to something like a circle? A ring could be helpful to see ordering if the player positions don't move from turn to turn. If so, there wouldn't need to be a change from vertical to horizontal

I think we should keep the left-to-right linear view available for a while, since it scales better for large games / large numbers of decks on small screens. Building a ring-variant play UI is probably worthwhile, but it's not something that people have been actively asking for yet.

IMO it would be confusing to have the players in a ring shape if the trick display isn't
